components/apimgt/org.wso2.carbon.apimgt.spec.parser/src/main/java/org/wso2/carbon/apimgt/spec/parser/definitions/AsyncApiParserUtil.java

```diff
@@ -108,4 +286,541 @@ public static ErrorItem addErrorToValidationResponse(
+        } else {
+            throw new APIManagementException("Unsupported AsyncAPI Channel",
+                    ExceptionCodes.ERROR_READING_ASYNCAPI_SPECIFICATION);
+        }
+    }
+
+    /**
+     * This method will create and provide the respective Async API Channel with data from based on the version
+     *
+     * @param channels String
+     * @return AsyncApiDocument
+     */
+    public static AsyncApi30Channel createChannel(AsyncApiChannels channels) throws APIManagementException {
+        if (channels instanceof AsyncApi30Channels) {
+            return ((AsyncApi30Channels) channels).createChannel();
+        } else {
+            throw new APIManagementException("Unsupported AsyncAPI Channel",
+                    ExceptionCodes.ERROR_READING_ASYNCAPI_SPECIFICATION);
+        }
+    }
+
+    /**
+     * This method will return a list of channel items from the respective Async API document
+     *
+     * @param asyncDocument AsyncApiDocument
+     * @return List<AsyncApiChannelItem>
+     */
+    public static List<AsyncApiChannelItem> getAllChannels(AsyncApiDocument asyncDocument) {
+        List<AsyncApiChannelItem> channels = new ArrayList<>();
+        AsyncApiChannels apiChannels = asyncDocument.getChannels();
+
+        if (apiChannels instanceof MappedNode) {
+            MappedNode<?> mappedChannels = (MappedNode<?>) apiChannels;
+            for (Object item : mappedChannels.getItems()) {
+                if (item instanceof AsyncApiChannelItem) {
+                    channels.add((AsyncApiChannelItem) item);
+                }
+            }
+        }
+        return channels;
+    }
+
+    /**
+     * This method will return schema from the parameter object for the respective Async API based on the version
+     *
+     * @param parameterObj Object
+     * @return Object
+     */
+    public static Object getSchemaFromParameter(Object parameterObj) {
+        if (parameterObj instanceof AsyncApi20Parameter) {
+            return ((AsyncApi20Parameter) parameterObj).getSchema();
+        } else if (parameterObj instanceof AsyncApi21Parameter) {
+            return ((AsyncApi21Parameter) parameterObj).getSchema();
+        } else if (parameterObj instanceof AsyncApi22Parameter) {
+            return ((AsyncApi22Parameter) parameterObj).getSchema();
+        } else if (parameterObj instanceof AsyncApi23Parameter) {
+            return ((AsyncApi23Parameter) parameterObj).getSchema();
+        } else if (parameterObj instanceof AsyncApi24Parameter) {
+            return ((AsyncApi24Parameter) parameterObj).getSchema();
+        } else if (parameterObj instanceof AsyncApi25Parameter) {
+            return ((AsyncApi25Parameter) parameterObj).getSchema();
+        } else if (parameterObj instanceof AsyncApi26Parameter) {
+            return ((AsyncApi26Parameter) parameterObj).getSchema();
+        } else {
+            throw new UnsupportedOperationException("Unsupported AsyncAPI Parameter");
```

Comment:
is this exception handled from calling method? shall we change it to APIManagementException